packages/quantic/force-app/main/default/lwc/quanticRecentResultsList/e2e/pageObject.ts

```diff
@@ -0,0 +1,70 @@
+import type {Locator, Page, Request} from '@playwright/test';
+import {isUaCustomEvent} from '../../../../../../playwright/utils/requests';
+
+export class RecentResultListObject {
+  constructor(public page: Page) {
+    this.page = page;
+  }
+
+  get recentResultList(): Locator {
+    return this.page.locator('c-quantic-recent-result-list');
+  }
+
+  get recentResultListItemLinks(): Locator {
+    return this.page.locator('c-quantic-recent-result-link');
+  }
+
+  get selectResultButton(): Locator {
+    return this.page.locator('c-action-select-results button');
+  }
+
+  async clickSelectResultButton(): Promise<void> {
+    await this.selectResultButton.click();
+  }
+
+  getRecentResultLinkByIndex(index: number): Locator {
+    return this.recentResultListItemLinks.nth(index);
+  }
+
+  async waitForRecentResultClickAnalytics(
+    expectedFields: Record<string, any>
+  ): Promise<Request> {
+    return this.waitForRecentResultListSearchAnalytics(
+      'recentlyClickedDocuments',
+      'recentResultClick',
+      expectedFields.documentTitle,
+      expectedFields.documentUrl
+    );
+  }
```

Comment:
For the analytics you can start using the new analytics logic presented in this PR: https://github.com/coveo/ui-kit/pull/5297